### PR TITLE
Add a common tags to all Kaldb metrics

### PR DESCRIPTION
**Pull request by mansu**
Add component name and common tags to all metrics so it's easier to slice and dice them in dashboards by cluster name. Adding new static cluster name and env configs so they can be passed in as configs.

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/server/Kaldb.java [hunk 1]
```diff
@@ -69,6 +72,16 @@ public class Kaldb {
...
 
   Kaldb(KaldbConfigs.KaldbConfig kaldbConfig, S3Client s3Client) {
+    // Configure prometheus meter registry
+    prometheusMeterRegistry
+        .config()
+        .meterFilter(
+            new MeterFilter() {
+              @Override
+              public Meter.Id map(Meter.Id id) {
+                return id.withName(KALDB_METRICS_PREFIX + id.getName());
```
Comment (vthacker) [inline kaldb/src/main/java/com/slack/kaldb/server/Kaldb.java:82]: What happens to metrics that we changed recently to manually have `kaldb_` prefix?

**Review by bryanlb — DISMISSED** [review]
There's a few issues with this approach as identified, but I think we probably don't want to proceed with this approach. The better way would be to fix the individual metrics that are missing prefixes - we already need to audit the naming so that they're more consistent.

File: kaldb/src/main/java/com/slack/kaldb/server/Kaldb.java [hunk 2]
```diff
@@ -57,6 +59,7 @@
  * http server, register monitoring libraries, create config manager etc..
  */
 public class Kaldb {
+  public static final String KALDB_METRICS_PREFIX = "kaldb.";
```
[the remark this answers is not in the document]
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/server/Kaldb.java]: Removed the prefix filter.

File: kaldb/src/main/java/com/slack/kaldb/server/Kaldb.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/server/Kaldb.java]: In general, this guidance applies to micro services, so we can graph the data from multiple services via a common dashboards. For a product like Kaldb, it's better to prefix them so we know which metrics are emitted by Kaldb. Either ways, removed the prefix filter for now so it's not a concern.

**Comment by mansu** [comment]
Updated the diff to just use common tags and not a common prefix.

File: config/config.yaml [hunk 3]
```diff
@@ -76,6 +76,10 @@ managerConfig:
     schedulePeriodMins: ${KALDB_MANAGER_SNAPSHOT_DELETE_PERIOD_MINS:-15}
     snapshotLifespanMins: ${KALDB_MANAGER_SNAPSHOT_LIFESPAN_MINS:-10080}
 
+clusterConfig:
+  clusterName: ${KALDB_CLUSTER_NAME:-kaldb_local}
+  env: ${KALDB_CLUSTER_ENV:-local}
```
[the remark this answers is not in the document]
Comment (mansu) [reply config/config.yaml:81]: local is a preferable default since dev could be a real env name.

**Review by vthacker — APPROVED** [review]

**Pull request merged**